src/main/scala/lenthall/spray/SprayCanHttpService.scala

```diff
@@ -0,0 +1,169 @@
+package lenthall.spray
+
+import java.net.{InetAddress, InetSocketAddress}
+
+import akka.actor._
+import akka.io.Tcp.Unbound
+import akka.io.{IO, Inet}
+import akka.util.Timeout
+import spray.can.Http
+import spray.can.Http.Bound
+import spray.can.server.ServerSettings
+import spray.io.ServerSSLEngineProvider
+
+import scala.collection.immutable
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ExecutionContext, Future, Promise, TimeoutException}
+
+/**
+ * Adds bind and unbind to http services running on spray-can.
+ *
+ * {{{
+ * val httpServiceActorRef: ActorRef = ...
+ * httpServiceActorRef.bind(LoopbackAddress, 8080) onSuccess {
+ *   case boundListener =>
+ *     boundListener.unbind() // Unbind the listener
+ * }
+ * }}}
+ */
+object SprayCanHttpService {
+
+  val LoopbackAddress = InetAddress.getLoopbackAddress.getHostAddress
+  val AnyAddress = new InetSocketAddress(0).getAddress.getHostAddress
+
+  implicit class EnhancedSprayCanActor(val service: ActorRef) extends AnyVal {
+
+    def bind(interface: String = LoopbackAddress, port: Int = 80, backlog: Int = 100,
+             options: immutable.Traversable[Inet.SocketOption] = Nil, settings: Option[ServerSettings] = None)
+            (implicit ec: ExecutionContext, timeout: Timeout, actorSystem: ActorSystem,
+             sslEngineProvider: ServerSSLEngineProvider): Future[BoundListener] = {
+      Future(Http.Bind(service, interface, port, backlog, options, settings)) flatMap {
+        case httpBind =>
+          val promise = Promise[BoundListener]()
+          actorSystem.actorOf(Props(classOf[SprayCanBindActor], httpBind, promise, timeout))
+          promise.future
+      }
+    }
+
+    def bindOrShutdown(interface: String = LoopbackAddress, port: Int = 80, backlog: Int = 100,
+                       options: immutable.Traversable[Inet.SocketOption] = Nil, settings: Option[ServerSettings] = None)
+                      (implicit ec: ExecutionContext, actorSystem: ActorSystem, timeout: Timeout,
+                       sslEngineProvider: ServerSSLEngineProvider): Future[BoundListener] = {
+      bind(interface, port, backlog, options, settings) recover {
+        case exception =>
+          actorSystem.log.error(exception, s"Binding failed interface $interface port $port")
+          actorSystem.shutdown()
+          actorSystem.awaitTermination()
+          throw exception
+      }
+    }
+
+  }
+
+  case class BoundListener(bound: Bound, listener: ActorRef) {
+
+    def unbind(duration: Duration = Duration.Zero)
+              (implicit ec: ExecutionContext, actorSystem: ActorSystem, timeout: Timeout): Future[Unbound] = {
+      Future(Http.Unbind(duration)) flatMap {
+        case httpUnbind =>
```

Comment:
I agree that `flatMap` would only be passed an `HttpUnbind`, I just wasn't sure why the `case` is here as it isn't putting any restrictions on what will be passed.